cornice/schemas.py

```diff
@@ -175,3 +175,78 @@ def _validate_fields(location, data):
         for param in set(params) - set([attr.name for attr in attrs]):
             request.errors.add('body' if param in body else 'querystring',
                                param, msg % param)
+
+
+def validate_matchdict(matchdict_types):
+    """Validates types of request.matchdict through matchdict_types
+    dictionary.
+
+    matchdict_types dictionary should have it's values defined as
+    callable type. Here's an example::
+
+        @view(
+            ...
+            validators=( validate_matchdict( {'id': int} ), ),
+            ...
+        )
+    """
+    def wrapper(request):
+        md = request.matchdict
+        for k, v in matchdict_types.iteritems():
+            if k in md.keys():
+                item = md.get(k)
+                try:
+                    v(item)
+                except Exception as e:
+                    request.errors.add('param', 'exception', str(e))
```

Comment:
This is not how these errors are supposed to be used. This should be something like `request.errors.add(location, name, description)`